conans/client/cmd/export.py

```diff
@@ -47,12 +48,24 @@ def check_casing_conflict(cache, ref):
                              % (str(ref), " ".join(str(s) for s in refs)))
 
 
-def cmd_export(package_layout, conanfile_path, conanfile, keep_source, revisions_enabled,
-               output, hook_manager):
+def cmd_export(conanfile_path, name, version, user, channel, keep_source, revisions_enabled,
+               output, hook_manager, loader, cache, export=True):
     """ Export the recipe
     param conanfile_path: the original source directory of the user containing a
                        conanfile.py
     """
+    conanfile = loader.load_export(conanfile_path, name, version, user, channel)
+    ref = ConanFileReference(conanfile.name, conanfile.version, conanfile.user,
+                             conanfile.channel)
+    check_casing_conflict(cache=cache, ref=ref)
```

Comment:
This was checked only `if not not_export` in the `create`, do we care?

Reply:
I think it is better to check, if the case doesn't match and you skip the check in the create, things might go very wrong.